File: 2015/1/main.py
f = open("input.txt", "r")
input = f.readline()
f.close()

floors = 0
first_basement = True

for i in range(0, len(input)):
    match input[i]:
        case "(":
            floors += 1
        case ")":
            floors -= 1

    if floors < 0 and first_basement is True:
        print(f"First basement entry: {i + 1}")
        first_basement = False

print(f"Floors: {floors}")

File: 2015/1/cpp/main.cpp
#include <fstream>
#include <iostream>
#include <string>

int main() {
  std::ifstream file;

  file.open("input.txt");

  std::string line = "";

  std::getline(file, line);

  int counter = 0;
  int i = 0;
  int basementFirstEntered = 0;
  bool basementEntered = false;

  for (char c : line) {
    switch (c) {
    case '(':
      counter++;
      break;
    case ')':
      counter--;
      break;
    }
    if (counter < 0 && !basementEntered) {
      basementFirstEntered = i + 1;
      basementEntered = true;
    }
    i++;
  }
  std::cout << "Floor: ";
  std::cout << counter;
  std::cout << "\n";
  std::cout << "Basement First Entered: ";
  std::cout << basementFirstEntered;
  std::cout << "\n";
}
